okio/src/main/java/okio/AsyncTimeout.java

```diff
@@ -151,6 +130,24 @@ private long remainingNanos(long now) {
   protected void timedOut() {
   }
 
+  @Override
+  public final long getDelay(TimeUnit unit) {
+    return unit.convert(timeoutAt - System.nanoTime(), TimeUnit.NANOSECONDS);
+  }
+
+  @Override
+  public final int compareTo(Delayed that) {
+    long thisDelay = getDelay(TimeUnit.NANOSECONDS);
```

Comment:
good call... i made it prefer we're an `AsyncTimeout` but fall back to comparing getDelay